{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "96423068",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tximport)\n",
    "library(DESeq2)\n",
    "library(apeglm)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "bafeb610",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'accession_log.txt'</li><li>'failed_acc.txt'</li><li>'fastq'</li><li>'filtered'</li><li>'filtered_logs'</li><li>'metadata.txt'</li><li>'NexteraPE-PE.fa'</li><li>'Pan_sequences.fa'</li><li>'README.md'</li><li>'rsem'</li><li>'sample_info.csv'</li><li>'SRR_Acc_List.txt'</li><li>'STAR'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'accession\\_log.txt'\n",
       "\\item 'failed\\_acc.txt'\n",
       "\\item 'fastq'\n",
       "\\item 'filtered'\n",
       "\\item 'filtered\\_logs'\n",
       "\\item 'metadata.txt'\n",
       "\\item 'NexteraPE-PE.fa'\n",
       "\\item 'Pan\\_sequences.fa'\n",
       "\\item 'README.md'\n",
       "\\item 'rsem'\n",
       "\\item 'sample\\_info.csv'\n",
       "\\item 'SRR\\_Acc\\_List.txt'\n",
       "\\item 'STAR'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'accession_log.txt'\n",
       "2. 'failed_acc.txt'\n",
       "3. 'fastq'\n",
       "4. 'filtered'\n",
       "5. 'filtered_logs'\n",
       "6. 'metadata.txt'\n",
       "7. 'NexteraPE-PE.fa'\n",
       "8. 'Pan_sequences.fa'\n",
       "9. 'README.md'\n",
       "10. 'rsem'\n",
       "11. 'sample_info.csv'\n",
       "12. 'SRR_Acc_List.txt'\n",
       "13. 'STAR'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"accession_log.txt\" \"failed_acc.txt\"    \"fastq\"            \n",
       " [4] \"filtered\"          \"filtered_logs\"     \"metadata.txt\"     \n",
       " [7] \"NexteraPE-PE.fa\"   \"Pan_sequences.fa\"  \"README.md\"        \n",
       "[10] \"rsem\"              \"sample_info.csv\"   \"SRR_Acc_List.txt\" \n",
       "[13] \"STAR\"             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "dir <- \"../data\"\n",
    "list.files(dir)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "820d0a27",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 6 × 26</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>Run</th><th scope=col>age</th><th scope=col>Assay.Type</th><th scope=col>AvgSpotLen</th><th scope=col>Bases</th><th scope=col>BioProject</th><th scope=col>BioSample</th><th scope=col>Bytes</th><th scope=col>Center.Name</th><th scope=col>Consent</th><th scope=col>⋯</th><th scope=col>LibraryLayout</th><th scope=col>LibrarySelection</th><th scope=col>LibrarySource</th><th scope=col>Organism</th><th scope=col>Platform</th><th scope=col>ReleaseDate</th><th scope=col>Sample.Name</th><th scope=col>source_name</th><th scope=col>SRA.Study</th><th scope=col>Strain</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>⋯</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>1</th><td>SRR10079579</td><td>2-month-old</td><td>RNA-Seq</td><td>300</td><td>8062648800</td><td>PRJNA564299</td><td>SAMN12708617</td><td>2635498394</td><td>GEO</td><td>public</td><td>⋯</td><td>PAIRED</td><td>cDNA</td><td>TRANSCRIPTOMIC</td><td>Mus musculus</td><td>ILLUMINA</td><td>2020-04-21T00:00:00Z</td><td>GSM4065891</td><td>Brain</td><td>SRP220678</td><td>C57BL/6</td></tr>\n",
       "\t<tr><th scope=row>2</th><td>SRR10079580</td><td>2-month-old</td><td>RNA-Seq</td><td>300</td><td>8013847800</td><td>PRJNA564299</td><td>SAMN12708616</td><td>2636901025</td><td>GEO</td><td>public</td><td>⋯</td><td>PAIRED</td><td>cDNA</td><td>TRANSCRIPTOMIC</td><td>Mus musculus</td><td>ILLUMINA</td><td>2020-04-21T00:00:00Z</td><td>GSM4065892</td><td>Brain</td><td>SRP220678</td><td>C57BL/6</td></tr>\n",
       "\t<tr><th scope=row>3</th><td>SRR10079581</td><td>2-month-old</td><td>RNA-Seq</td><td>300</td><td>8055843600</td><td>PRJNA564299</td><td>SAMN12708615</td><td>2642059362</td><td>GEO</td><td>public</td><td>⋯</td><td>PAIRED</td><td>cDNA</td><td>TRANSCRIPTOMIC</td><td>Mus musculus</td><td>ILLUMINA</td><td>2020-04-21T00:00:00Z</td><td>GSM4065893</td><td>Brain</td><td>SRP220678</td><td>C57BL/6</td></tr>\n",
       "\t<tr><th scope=row>4</th><td>SRR10079582</td><td>4-month-old</td><td>RNA-Seq</td><td>300</td><td>7608938400</td><td>PRJNA564299</td><td>SAMN12708614</td><td>2465813362</td><td>GEO</td><td>public</td><td>⋯</td><td>PAIRED</td><td>cDNA</td><td>TRANSCRIPTOMIC</td><td>Mus musculus</td><td>ILLUMINA</td><td>2020-04-21T00:00:00Z</td><td>GSM4065894</td><td>Brain</td><td>SRP220678</td><td>C57BL/6</td></tr>\n",
       "\t<tr><th scope=row>5</th><td>SRR10079583</td><td>4-month-old</td><td>RNA-Seq</td><td>300</td><td>7912941900</td><td>PRJNA564299</td><td>SAMN12708613</td><td>2604058160</td><td>GEO</td><td>public</td><td>⋯</td><td>PAIRED</td><td>cDNA</td><td>TRANSCRIPTOMIC</td><td>Mus musculus</td><td>ILLUMINA</td><td>2020-04-21T00:00:00Z</td><td>GSM4065895</td><td>Brain</td><td>SRP220678</td><td>C57BL/6</td></tr>\n",
       "\t<tr><th scope=row>6</th><td>SRR10079584</td><td>4-month-old</td><td>RNA-Seq</td><td>300</td><td>7563385200</td><td>PRJNA564299</td><td>SAMN12708612</td><td>2455545524</td><td>GEO</td><td>public</td><td>⋯</td><td>PAIRED</td><td>cDNA</td><td>TRANSCRIPTOMIC</td><td>Mus musculus</td><td>ILLUMINA</td><td>2020-04-21T00:00:00Z</td><td>GSM4065896</td><td>Brain</td><td>SRP220678</td><td>C57BL/6</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 6 × 26\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & Run & age & Assay.Type & AvgSpotLen & Bases & BioProject & BioSample & Bytes & Center.Name & Consent & ⋯ & LibraryLayout & LibrarySelection & LibrarySource & Organism & Platform & ReleaseDate & Sample.Name & source\\_name & SRA.Study & Strain\\\\\n",
       "  & <chr> & <chr> & <chr> & <int> & <dbl> & <chr> & <chr> & <dbl> & <chr> & <chr> & ⋯ & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t1 & SRR10079579 & 2-month-old & RNA-Seq & 300 & 8062648800 & PRJNA564299 & SAMN12708617 & 2635498394 & GEO & public & ⋯ & PAIRED & cDNA & TRANSCRIPTOMIC & Mus musculus & ILLUMINA & 2020-04-21T00:00:00Z & GSM4065891 & Brain & SRP220678 & C57BL/6\\\\\n",
       "\t2 & SRR10079580 & 2-month-old & RNA-Seq & 300 & 8013847800 & PRJNA564299 & SAMN12708616 & 2636901025 & GEO & public & ⋯ & PAIRED & cDNA & TRANSCRIPTOMIC & Mus musculus & ILLUMINA & 2020-04-21T00:00:00Z & GSM4065892 & Brain & SRP220678 & C57BL/6\\\\\n",
       "\t3 & SRR10079581 & 2-month-old & RNA-Seq & 300 & 8055843600 & PRJNA564299 & SAMN12708615 & 2642059362 & GEO & public & ⋯ & PAIRED & cDNA & TRANSCRIPTOMIC & Mus musculus & ILLUMINA & 2020-04-21T00:00:00Z & GSM4065893 & Brain & SRP220678 & C57BL/6\\\\\n",
       "\t4 & SRR10079582 & 4-month-old & RNA-Seq & 300 & 7608938400 & PRJNA564299 & SAMN12708614 & 2465813362 & GEO & public & ⋯ & PAIRED & cDNA & TRANSCRIPTOMIC & Mus musculus & ILLUMINA & 2020-04-21T00:00:00Z & GSM4065894 & Brain & SRP220678 & C57BL/6\\\\\n",
       "\t5 & SRR10079583 & 4-month-old & RNA-Seq & 300 & 7912941900 & PRJNA564299 & SAMN12708613 & 2604058160 & GEO & public & ⋯ & PAIRED & cDNA & TRANSCRIPTOMIC & Mus musculus & ILLUMINA & 2020-04-21T00:00:00Z & GSM4065895 & Brain & SRP220678 & C57BL/6\\\\\n",
       "\t6 & SRR10079584 & 4-month-old & RNA-Seq & 300 & 7563385200 & PRJNA564299 & SAMN12708612 & 2455545524 & GEO & public & ⋯ & PAIRED & cDNA & TRANSCRIPTOMIC & Mus musculus & ILLUMINA & 2020-04-21T00:00:00Z & GSM4065896 & Brain & SRP220678 & C57BL/6\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 6 × 26\n",
       "\n",
       "| <!--/--> | Run &lt;chr&gt; | age &lt;chr&gt; | Assay.Type &lt;chr&gt; | AvgSpotLen &lt;int&gt; | Bases &lt;dbl&gt; | BioProject &lt;chr&gt; | BioSample &lt;chr&gt; | Bytes &lt;dbl&gt; | Center.Name &lt;chr&gt; | Consent &lt;chr&gt; | ⋯ ⋯ | LibraryLayout &lt;chr&gt; | LibrarySelection &lt;chr&gt; | LibrarySource &lt;chr&gt; | Organism &lt;chr&gt; | Platform &lt;chr&gt; | ReleaseDate &lt;chr&gt; | Sample.Name &lt;chr&gt; | source_name &lt;chr&gt; | SRA.Study &lt;chr&gt; | Strain &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 1 | SRR10079579 | 2-month-old | RNA-Seq | 300 | 8062648800 | PRJNA564299 | SAMN12708617 | 2635498394 | GEO | public | ⋯ | PAIRED | cDNA | TRANSCRIPTOMIC | Mus musculus | ILLUMINA | 2020-04-21T00:00:00Z | GSM4065891 | Brain | SRP220678 | C57BL/6 |\n",
       "| 2 | SRR10079580 | 2-month-old | RNA-Seq | 300 | 8013847800 | PRJNA564299 | SAMN12708616 | 2636901025 | GEO | public | ⋯ | PAIRED | cDNA | TRANSCRIPTOMIC | Mus musculus | ILLUMINA | 2020-04-21T00:00:00Z | GSM4065892 | Brain | SRP220678 | C57BL/6 |\n",
       "| 3 | SRR10079581 | 2-month-old | RNA-Seq | 300 | 8055843600 | PRJNA564299 | SAMN12708615 | 2642059362 | GEO | public | ⋯ | PAIRED | cDNA | TRANSCRIPTOMIC | Mus musculus | ILLUMINA | 2020-04-21T00:00:00Z | GSM4065893 | Brain | SRP220678 | C57BL/6 |\n",
       "| 4 | SRR10079582 | 4-month-old | RNA-Seq | 300 | 7608938400 | PRJNA564299 | SAMN12708614 | 2465813362 | GEO | public | ⋯ | PAIRED | cDNA | TRANSCRIPTOMIC | Mus musculus | ILLUMINA | 2020-04-21T00:00:00Z | GSM4065894 | Brain | SRP220678 | C57BL/6 |\n",
       "| 5 | SRR10079583 | 4-month-old | RNA-Seq | 300 | 7912941900 | PRJNA564299 | SAMN12708613 | 2604058160 | GEO | public | ⋯ | PAIRED | cDNA | TRANSCRIPTOMIC | Mus musculus | ILLUMINA | 2020-04-21T00:00:00Z | GSM4065895 | Brain | SRP220678 | C57BL/6 |\n",
       "| 6 | SRR10079584 | 4-month-old | RNA-Seq | 300 | 7563385200 | PRJNA564299 | SAMN12708612 | 2455545524 | GEO | public | ⋯ | PAIRED | cDNA | TRANSCRIPTOMIC | Mus musculus | ILLUMINA | 2020-04-21T00:00:00Z | GSM4065896 | Brain | SRP220678 | C57BL/6 |\n",
       "\n"
      ],
      "text/plain": [
       "  Run         age         Assay.Type AvgSpotLen Bases      BioProject \n",
       "1 SRR10079579 2-month-old RNA-Seq    300        8062648800 PRJNA564299\n",
       "2 SRR10079580 2-month-old RNA-Seq    300        8013847800 PRJNA564299\n",
       "3 SRR10079581 2-month-old RNA-Seq    300        8055843600 PRJNA564299\n",
       "4 SRR10079582 4-month-old RNA-Seq    300        7608938400 PRJNA564299\n",
       "5 SRR10079583 4-month-old RNA-Seq    300        7912941900 PRJNA564299\n",
       "6 SRR10079584 4-month-old RNA-Seq    300        7563385200 PRJNA564299\n",
       "  BioSample    Bytes      Center.Name Consent ⋯ LibraryLayout LibrarySelection\n",
       "1 SAMN12708617 2635498394 GEO         public  ⋯ PAIRED        cDNA            \n",
       "2 SAMN12708616 2636901025 GEO         public  ⋯ PAIRED        cDNA            \n",
       "3 SAMN12708615 2642059362 GEO         public  ⋯ PAIRED        cDNA            \n",
       "4 SAMN12708614 2465813362 GEO         public  ⋯ PAIRED        cDNA            \n",
       "5 SAMN12708613 2604058160 GEO         public  ⋯ PAIRED        cDNA            \n",
       "6 SAMN12708612 2455545524 GEO         public  ⋯ PAIRED        cDNA            \n",
       "  LibrarySource  Organism     Platform ReleaseDate          Sample.Name\n",
       "1 TRANSCRIPTOMIC Mus musculus ILLUMINA 2020-04-21T00:00:00Z GSM4065891 \n",
       "2 TRANSCRIPTOMIC Mus musculus ILLUMINA 2020-04-21T00:00:00Z GSM4065892 \n",
       "3 TRANSCRIPTOMIC Mus musculus ILLUMINA 2020-04-21T00:00:00Z GSM4065893 \n",
       "4 TRANSCRIPTOMIC Mus musculus ILLUMINA 2020-04-21T00:00:00Z GSM4065894 \n",
       "5 TRANSCRIPTOMIC Mus musculus ILLUMINA 2020-04-21T00:00:00Z GSM4065895 \n",
       "6 TRANSCRIPTOMIC Mus musculus ILLUMINA 2020-04-21T00:00:00Z GSM4065896 \n",
       "  source_name SRA.Study Strain \n",
       "1 Brain       SRP220678 C57BL/6\n",
       "2 Brain       SRP220678 C57BL/6\n",
       "3 Brain       SRP220678 C57BL/6\n",
       "4 Brain       SRP220678 C57BL/6\n",
       "5 Brain       SRP220678 C57BL/6\n",
       "6 Brain       SRP220678 C57BL/6"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "samples <- read.table(file.path(dir, \"metadata.txt\"), sep = \",\", header = TRUE)\n",
    "head(samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "8fd1bc6c",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "reading in files with read_tsv\n",
      "\n",
      "1 \n",
      "2 \n",
      "3 \n",
      "4 \n",
      "5 \n",
      "6 \n",
      "7 \n",
      "8 \n",
      "9 \n",
      "10 \n",
      "11 \n",
      "12 \n",
      "13 \n",
      "14 \n",
      "15 \n",
      "16 \n",
      "17 \n",
      "18 \n",
      "19 \n",
      "20 \n",
      "21 \n",
      "22 \n",
      "23 \n",
      "24 \n",
      "25 \n",
      "26 \n",
      "27 \n",
      "28 \n",
      "29 \n",
      "30 \n",
      "31 \n",
      "32 \n",
      "33 \n",
      "34 \n",
      "35 \n",
      "36 \n",
      "37 \n",
      "38 \n",
      "39 \n",
      "40 \n",
      "41 \n",
      "42 \n",
      "43 \n",
      "44 \n",
      "45 \n",
      "46 \n",
      "47 \n",
      "48 \n",
      "49 \n",
      "50 \n",
      "51 \n",
      "52 \n",
      "53 \n",
      "54 \n",
      "55 \n",
      "56 \n",
      "57 \n",
      "58 \n",
      "59 \n",
      "60 \n",
      "\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 6 × 60 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample1</th><th scope=col>sample2</th><th scope=col>sample3</th><th scope=col>sample4</th><th scope=col>sample5</th><th scope=col>sample6</th><th scope=col>sample7</th><th scope=col>sample8</th><th scope=col>sample9</th><th scope=col>sample10</th><th scope=col>⋯</th><th scope=col>sample51</th><th scope=col>sample52</th><th scope=col>sample53</th><th scope=col>sample54</th><th scope=col>sample55</th><th scope=col>sample56</th><th scope=col>sample57</th><th scope=col>sample58</th><th scope=col>sample59</th><th scope=col>sample60</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>0610005C13Rik</th><td>  0</td><td>  0</td><td>  1</td><td>  1</td><td>  1</td><td>  0</td><td>  1</td><td>  0</td><td>  1</td><td>  0</td><td>⋯</td><td>   2</td><td>   7</td><td>   6</td><td>   3</td><td>  2</td><td>   0</td><td>   0</td><td>   9</td><td>   6</td><td>   4</td></tr>\n",
       "\t<tr><th scope=row>0610006L08Rik</th><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>⋯</td><td>   0</td><td>   1</td><td>   0</td><td>   1</td><td>  0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>0610007P14Rik</th><td>327</td><td>360</td><td>442</td><td>358</td><td>429</td><td>347</td><td>572</td><td>585</td><td>697</td><td>408</td><td>⋯</td><td>1214</td><td>1667</td><td>1672</td><td>1704</td><td>972</td><td>1167</td><td>1294</td><td>1855</td><td>1509</td><td>1333</td></tr>\n",
       "\t<tr><th scope=row>0610009B22Rik</th><td>631</td><td>604</td><td>834</td><td>471</td><td>614</td><td>497</td><td>926</td><td>957</td><td>929</td><td>752</td><td>⋯</td><td> 735</td><td>1006</td><td>1045</td><td>1054</td><td>801</td><td> 901</td><td> 855</td><td>1180</td><td>1002</td><td> 895</td></tr>\n",
       "\t<tr><th scope=row>0610009E02Rik</th><td>  2</td><td>  2</td><td>  2</td><td>  3</td><td>  4</td><td>  4</td><td>  1</td><td>  5</td><td>  5</td><td>  0</td><td>⋯</td><td>  22</td><td>  12</td><td>  28</td><td>  18</td><td> 11</td><td>   7</td><td>   6</td><td>  22</td><td>  21</td><td>   7</td></tr>\n",
       "\t<tr><th scope=row>0610009L18Rik</th><td> 32</td><td> 64</td><td> 34</td><td> 48</td><td> 57</td><td> 28</td><td> 71</td><td> 53</td><td> 62</td><td> 48</td><td>⋯</td><td> 148</td><td> 117</td><td> 194</td><td>  88</td><td> 93</td><td> 114</td><td> 128</td><td> 184</td><td> 187</td><td> 125</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 6 × 60 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllllllllll}\n",
       "  & sample1 & sample2 & sample3 & sample4 & sample5 & sample6 & sample7 & sample8 & sample9 & sample10 & ⋯ & sample51 & sample52 & sample53 & sample54 & sample55 & sample56 & sample57 & sample58 & sample59 & sample60\\\\\n",
       "\\hline\n",
       "\t0610005C13Rik &   0 &   0 &   1 &   1 &   1 &   0 &   1 &   0 &   1 &   0 & ⋯ &    2 &    7 &    6 &    3 &   2 &    0 &    0 &    9 &    6 &    4\\\\\n",
       "\t0610006L08Rik &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 & ⋯ &    0 &    1 &    0 &    1 &   0 &    0 &    0 &    0 &    0 &    0\\\\\n",
       "\t0610007P14Rik & 327 & 360 & 442 & 358 & 429 & 347 & 572 & 585 & 697 & 408 & ⋯ & 1214 & 1667 & 1672 & 1704 & 972 & 1167 & 1294 & 1855 & 1509 & 1333\\\\\n",
       "\t0610009B22Rik & 631 & 604 & 834 & 471 & 614 & 497 & 926 & 957 & 929 & 752 & ⋯ &  735 & 1006 & 1045 & 1054 & 801 &  901 &  855 & 1180 & 1002 &  895\\\\\n",
       "\t0610009E02Rik &   2 &   2 &   2 &   3 &   4 &   4 &   1 &   5 &   5 &   0 & ⋯ &   22 &   12 &   28 &   18 &  11 &    7 &    6 &   22 &   21 &    7\\\\\n",
       "\t0610009L18Rik &  32 &  64 &  34 &  48 &  57 &  28 &  71 &  53 &  62 &  48 & ⋯ &  148 &  117 &  194 &   88 &  93 &  114 &  128 &  184 &  187 &  125\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 6 × 60 of type dbl\n",
       "\n",
       "| <!--/--> | sample1 | sample2 | sample3 | sample4 | sample5 | sample6 | sample7 | sample8 | sample9 | sample10 | ⋯ | sample51 | sample52 | sample53 | sample54 | sample55 | sample56 | sample57 | sample58 | sample59 | sample60 |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 0610005C13Rik |   0 |   0 |   1 |   1 |   1 |   0 |   1 |   0 |   1 |   0 | ⋯ |    2 |    7 |    6 |    3 |   2 |    0 |    0 |    9 |    6 |    4 |\n",
       "| 0610006L08Rik |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 | ⋯ |    0 |    1 |    0 |    1 |   0 |    0 |    0 |    0 |    0 |    0 |\n",
       "| 0610007P14Rik | 327 | 360 | 442 | 358 | 429 | 347 | 572 | 585 | 697 | 408 | ⋯ | 1214 | 1667 | 1672 | 1704 | 972 | 1167 | 1294 | 1855 | 1509 | 1333 |\n",
       "| 0610009B22Rik | 631 | 604 | 834 | 471 | 614 | 497 | 926 | 957 | 929 | 752 | ⋯ |  735 | 1006 | 1045 | 1054 | 801 |  901 |  855 | 1180 | 1002 |  895 |\n",
       "| 0610009E02Rik |   2 |   2 |   2 |   3 |   4 |   4 |   1 |   5 |   5 |   0 | ⋯ |   22 |   12 |   28 |   18 |  11 |    7 |    6 |   22 |   21 |    7 |\n",
       "| 0610009L18Rik |  32 |  64 |  34 |  48 |  57 |  28 |  71 |  53 |  62 |  48 | ⋯ |  148 |  117 |  194 |   88 |  93 |  114 |  128 |  184 |  187 |  125 |\n",
       "\n"
      ],
      "text/plain": [
       "              sample1 sample2 sample3 sample4 sample5 sample6 sample7 sample8\n",
       "0610005C13Rik   0       0       1       1       1       0       1       0    \n",
       "0610006L08Rik   0       0       0       0       0       0       0       0    \n",
       "0610007P14Rik 327     360     442     358     429     347     572     585    \n",
       "0610009B22Rik 631     604     834     471     614     497     926     957    \n",
       "0610009E02Rik   2       2       2       3       4       4       1       5    \n",
       "0610009L18Rik  32      64      34      48      57      28      71      53    \n",
       "              sample9 sample10 ⋯ sample51 sample52 sample53 sample54 sample55\n",
       "0610005C13Rik   1       0      ⋯    2        7        6        3       2     \n",
       "0610006L08Rik   0       0      ⋯    0        1        0        1       0     \n",
       "0610007P14Rik 697     408      ⋯ 1214     1667     1672     1704     972     \n",
       "0610009B22Rik 929     752      ⋯  735     1006     1045     1054     801     \n",
       "0610009E02Rik   5       0      ⋯   22       12       28       18      11     \n",
       "0610009L18Rik  62      48      ⋯  148      117      194       88      93     \n",
       "              sample56 sample57 sample58 sample59 sample60\n",
       "0610005C13Rik    0        0        9        6        4    \n",
       "0610006L08Rik    0        0        0        0        0    \n",
       "0610007P14Rik 1167     1294     1855     1509     1333    \n",
       "0610009B22Rik  901      855     1180     1002      895    \n",
       "0610009E02Rik    7        6       22       21        7    \n",
       "0610009L18Rik  114      128      184      187      125    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "files <- file.path(dir, \"rsem\", samples$Run, paste0(samples$Run, \".genes.results\"))\n",
    "names(files) <- paste0(\"sample\", 1:60)\n",
    "txi <- tximport(files, type = \"rsem\", txIn = FALSE, txOut = FALSE)\n",
    "head(txi$counts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "6e1cc27c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 6 × 15 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample1</th><th scope=col>sample2</th><th scope=col>sample3</th><th scope=col>sample4</th><th scope=col>sample5</th><th scope=col>sample6</th><th scope=col>sample7</th><th scope=col>sample8</th><th scope=col>sample9</th><th scope=col>sample10</th><th scope=col>sample11</th><th scope=col>sample12</th><th scope=col>sample13</th><th scope=col>sample14</th><th scope=col>sample15</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>0610005C13Rik</th><td>  0</td><td>  0</td><td>  1</td><td>  1</td><td>  1</td><td>  0</td><td>  1</td><td>  0</td><td>  1</td><td>  0</td><td>  1</td><td>  0</td><td>  0</td><td>  0</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>0610006L08Rik</th><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>  0</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>0610007P14Rik</th><td>327</td><td>360</td><td>442</td><td>358</td><td>429</td><td>347</td><td>572</td><td>585</td><td>697</td><td>408</td><td>357</td><td>412</td><td>464</td><td>511</td><td> 195</td></tr>\n",
       "\t<tr><th scope=row>0610009B22Rik</th><td>631</td><td>604</td><td>834</td><td>471</td><td>614</td><td>497</td><td>926</td><td>957</td><td>929</td><td>752</td><td>675</td><td>679</td><td>518</td><td>718</td><td>2862</td></tr>\n",
       "\t<tr><th scope=row>0610009E02Rik</th><td>  2</td><td>  2</td><td>  2</td><td>  3</td><td>  4</td><td>  4</td><td>  1</td><td>  5</td><td>  5</td><td>  0</td><td>  0</td><td>  2</td><td>  3</td><td>  0</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>0610009L18Rik</th><td> 32</td><td> 64</td><td> 34</td><td> 48</td><td> 57</td><td> 28</td><td> 71</td><td> 53</td><td> 62</td><td> 48</td><td> 38</td><td> 54</td><td> 25</td><td> 26</td><td>   0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 6 × 15 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & sample1 & sample2 & sample3 & sample4 & sample5 & sample6 & sample7 & sample8 & sample9 & sample10 & sample11 & sample12 & sample13 & sample14 & sample15\\\\\n",
       "\\hline\n",
       "\t0610005C13Rik &   0 &   0 &   1 &   1 &   1 &   0 &   1 &   0 &   1 &   0 &   1 &   0 &   0 &   0 &    0\\\\\n",
       "\t0610006L08Rik &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &   0 &    0\\\\\n",
       "\t0610007P14Rik & 327 & 360 & 442 & 358 & 429 & 347 & 572 & 585 & 697 & 408 & 357 & 412 & 464 & 511 &  195\\\\\n",
       "\t0610009B22Rik & 631 & 604 & 834 & 471 & 614 & 497 & 926 & 957 & 929 & 752 & 675 & 679 & 518 & 718 & 2862\\\\\n",
       "\t0610009E02Rik &   2 &   2 &   2 &   3 &   4 &   4 &   1 &   5 &   5 &   0 &   0 &   2 &   3 &   0 &    0\\\\\n",
       "\t0610009L18Rik &  32 &  64 &  34 &  48 &  57 &  28 &  71 &  53 &  62 &  48 &  38 &  54 &  25 &  26 &    0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 6 × 15 of type dbl\n",
       "\n",
       "| <!--/--> | sample1 | sample2 | sample3 | sample4 | sample5 | sample6 | sample7 | sample8 | sample9 | sample10 | sample11 | sample12 | sample13 | sample14 | sample15 |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 0610005C13Rik |   0 |   0 |   1 |   1 |   1 |   0 |   1 |   0 |   1 |   0 |   1 |   0 |   0 |   0 |    0 |\n",
       "| 0610006L08Rik |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |   0 |    0 |\n",
       "| 0610007P14Rik | 327 | 360 | 442 | 358 | 429 | 347 | 572 | 585 | 697 | 408 | 357 | 412 | 464 | 511 |  195 |\n",
       "| 0610009B22Rik | 631 | 604 | 834 | 471 | 614 | 497 | 926 | 957 | 929 | 752 | 675 | 679 | 518 | 718 | 2862 |\n",
       "| 0610009E02Rik |   2 |   2 |   2 |   3 |   4 |   4 |   1 |   5 |   5 |   0 |   0 |   2 |   3 |   0 |    0 |\n",
       "| 0610009L18Rik |  32 |  64 |  34 |  48 |  57 |  28 |  71 |  53 |  62 |  48 |  38 |  54 |  25 |  26 |    0 |\n",
       "\n"
      ],
      "text/plain": [
       "              sample1 sample2 sample3 sample4 sample5 sample6 sample7 sample8\n",
       "0610005C13Rik   0       0       1       1       1       0       1       0    \n",
       "0610006L08Rik   0       0       0       0       0       0       0       0    \n",
       "0610007P14Rik 327     360     442     358     429     347     572     585    \n",
       "0610009B22Rik 631     604     834     471     614     497     926     957    \n",
       "0610009E02Rik   2       2       2       3       4       4       1       5    \n",
       "0610009L18Rik  32      64      34      48      57      28      71      53    \n",
       "              sample9 sample10 sample11 sample12 sample13 sample14 sample15\n",
       "0610005C13Rik   1       0        1        0        0        0         0    \n",
       "0610006L08Rik   0       0        0        0        0        0         0    \n",
       "0610007P14Rik 697     408      357      412      464      511       195    \n",
       "0610009B22Rik 929     752      675      679      518      718      2862    \n",
       "0610009E02Rik   5       0        0        2        3        0         0    \n",
       "0610009L18Rik  62      48       38       54       25       26         0    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A matrix: 6 × 15 of type dbl</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>sample31</th><th scope=col>sample32</th><th scope=col>sample33</th><th scope=col>sample34</th><th scope=col>sample35</th><th scope=col>sample36</th><th scope=col>sample37</th><th scope=col>sample38</th><th scope=col>sample39</th><th scope=col>sample40</th><th scope=col>sample41</th><th scope=col>sample42</th><th scope=col>sample43</th><th scope=col>sample44</th><th scope=col>sample45</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>0610005C13Rik</th><td>   0</td><td>   1</td><td>   0</td><td>   7</td><td>   7</td><td>   5</td><td>   6</td><td>   3</td><td>   3</td><td>   3</td><td>   3</td><td>   1</td><td>   7</td><td>   4</td><td>   3</td></tr>\n",
       "\t<tr><th scope=row>0610006L08Rik</th><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   0</td><td>   1</td><td>   0</td><td>   0</td></tr>\n",
       "\t<tr><th scope=row>0610007P14Rik</th><td>1139</td><td>1211</td><td>1403</td><td>1222</td><td>1087</td><td>1506</td><td>1987</td><td>2075</td><td>2088</td><td>1180</td><td>1216</td><td>1430</td><td>1341</td><td>1548</td><td>1378</td></tr>\n",
       "\t<tr><th scope=row>0610009B22Rik</th><td> 840</td><td> 873</td><td>1007</td><td> 736</td><td> 691</td><td> 901</td><td>1454</td><td>1432</td><td>1414</td><td> 893</td><td> 989</td><td> 997</td><td> 684</td><td> 722</td><td> 772</td></tr>\n",
       "\t<tr><th scope=row>0610009E02Rik</th><td>  10</td><td>   4</td><td>   2</td><td>  17</td><td>   6</td><td>  19</td><td>  29</td><td>  27</td><td>  28</td><td>   2</td><td>   6</td><td>   8</td><td>  20</td><td>  16</td><td>  15</td></tr>\n",
       "\t<tr><th scope=row>0610009L18Rik</th><td>  75</td><td>  59</td><td>  72</td><td> 129</td><td>  91</td><td> 104</td><td> 168</td><td> 177</td><td> 195</td><td> 119</td><td> 119</td><td> 132</td><td> 149</td><td> 131</td><td> 105</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A matrix: 6 × 15 of type dbl\n",
       "\\begin{tabular}{r|lllllllllllllll}\n",
       "  & sample31 & sample32 & sample33 & sample34 & sample35 & sample36 & sample37 & sample38 & sample39 & sample40 & sample41 & sample42 & sample43 & sample44 & sample45\\\\\n",
       "\\hline\n",
       "\t0610005C13Rik &    0 &    1 &    0 &    7 &    7 &    5 &    6 &    3 &    3 &    3 &    3 &    1 &    7 &    4 &    3\\\\\n",
       "\t0610006L08Rik &    0 &    0 &    0 &    0 &    0 &    0 &    0 &    0 &    0 &    0 &    0 &    0 &    1 &    0 &    0\\\\\n",
       "\t0610007P14Rik & 1139 & 1211 & 1403 & 1222 & 1087 & 1506 & 1987 & 2075 & 2088 & 1180 & 1216 & 1430 & 1341 & 1548 & 1378\\\\\n",
       "\t0610009B22Rik &  840 &  873 & 1007 &  736 &  691 &  901 & 1454 & 1432 & 1414 &  893 &  989 &  997 &  684 &  722 &  772\\\\\n",
       "\t0610009E02Rik &   10 &    4 &    2 &   17 &    6 &   19 &   29 &   27 &   28 &    2 &    6 &    8 &   20 &   16 &   15\\\\\n",
       "\t0610009L18Rik &   75 &   59 &   72 &  129 &   91 &  104 &  168 &  177 &  195 &  119 &  119 &  132 &  149 &  131 &  105\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A matrix: 6 × 15 of type dbl\n",
       "\n",
       "| <!--/--> | sample31 | sample32 | sample33 | sample34 | sample35 | sample36 | sample37 | sample38 | sample39 | sample40 | sample41 | sample42 | sample43 | sample44 | sample45 |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 0610005C13Rik |    0 |    1 |    0 |    7 |    7 |    5 |    6 |    3 |    3 |    3 |    3 |    1 |    7 |    4 |    3 |\n",
       "| 0610006L08Rik |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    0 |    1 |    0 |    0 |\n",
       "| 0610007P14Rik | 1139 | 1211 | 1403 | 1222 | 1087 | 1506 | 1987 | 2075 | 2088 | 1180 | 1216 | 1430 | 1341 | 1548 | 1378 |\n",
       "| 0610009B22Rik |  840 |  873 | 1007 |  736 |  691 |  901 | 1454 | 1432 | 1414 |  893 |  989 |  997 |  684 |  722 |  772 |\n",
       "| 0610009E02Rik |   10 |    4 |    2 |   17 |    6 |   19 |   29 |   27 |   28 |    2 |    6 |    8 |   20 |   16 |   15 |\n",
       "| 0610009L18Rik |   75 |   59 |   72 |  129 |   91 |  104 |  168 |  177 |  195 |  119 |  119 |  132 |  149 |  131 |  105 |\n",
       "\n"
      ],
      "text/plain": [
       "              sample31 sample32 sample33 sample34 sample35 sample36 sample37\n",
       "0610005C13Rik    0        1        0        7        7        5        6    \n",
       "0610006L08Rik    0        0        0        0        0        0        0    \n",
       "0610007P14Rik 1139     1211     1403     1222     1087     1506     1987    \n",
       "0610009B22Rik  840      873     1007      736      691      901     1454    \n",
       "0610009E02Rik   10        4        2       17        6       19       29    \n",
       "0610009L18Rik   75       59       72      129       91      104      168    \n",
       "              sample38 sample39 sample40 sample41 sample42 sample43 sample44\n",
       "0610005C13Rik    3        3        3        3        1        7        4    \n",
       "0610006L08Rik    0        0        0        0        0        1        0    \n",
       "0610007P14Rik 2075     2088     1180     1216     1430     1341     1548    \n",
       "0610009B22Rik 1432     1414      893      989      997      684      722    \n",
       "0610009E02Rik   27       28        2        6        8       20       16    \n",
       "0610009L18Rik  177      195      119      119      132      149      131    \n",
       "              sample45\n",
       "0610005C13Rik    3    \n",
       "0610006L08Rik    0    \n",
       "0610007P14Rik 1378    \n",
       "0610009B22Rik  772    \n",
       "0610009E02Rik   15    \n",
       "0610009L18Rik  105    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# WT microglia samples\n",
    "micro <- txi$counts[,1:15]\n",
    "head(micro)\n",
    "\n",
    "# WT astrocyte samples\n",
    "astro <- txi$counts[,31:45]\n",
    "head(astro)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "65a8ac4f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 15 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>condition</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>sample1</th><td>2mo </td></tr>\n",
       "\t<tr><th scope=row>sample2</th><td>2mo </td></tr>\n",
       "\t<tr><th scope=row>sample3</th><td>2mo </td></tr>\n",
       "\t<tr><th scope=row>sample4</th><td>4mo </td></tr>\n",
       "\t<tr><th scope=row>sample5</th><td>4mo </td></tr>\n",
       "\t<tr><th scope=row>sample6</th><td>4mo </td></tr>\n",
       "\t<tr><th scope=row>sample7</th><td>6mo </td></tr>\n",
       "\t<tr><th scope=row>sample8</th><td>6mo </td></tr>\n",
       "\t<tr><th scope=row>sample9</th><td>6mo </td></tr>\n",
       "\t<tr><th scope=row>sample10</th><td>9mo </td></tr>\n",
       "\t<tr><th scope=row>sample11</th><td>9mo </td></tr>\n",
       "\t<tr><th scope=row>sample12</th><td>9mo </td></tr>\n",
       "\t<tr><th scope=row>sample13</th><td>12mo</td></tr>\n",
       "\t<tr><th scope=row>sample14</th><td>12mo</td></tr>\n",
       "\t<tr><th scope=row>sample15</th><td>12mo</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 15 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & condition\\\\\n",
       "  & <fct>\\\\\n",
       "\\hline\n",
       "\tsample1 & 2mo \\\\\n",
       "\tsample2 & 2mo \\\\\n",
       "\tsample3 & 2mo \\\\\n",
       "\tsample4 & 4mo \\\\\n",
       "\tsample5 & 4mo \\\\\n",
       "\tsample6 & 4mo \\\\\n",
       "\tsample7 & 6mo \\\\\n",
       "\tsample8 & 6mo \\\\\n",
       "\tsample9 & 6mo \\\\\n",
       "\tsample10 & 9mo \\\\\n",
       "\tsample11 & 9mo \\\\\n",
       "\tsample12 & 9mo \\\\\n",
       "\tsample13 & 12mo\\\\\n",
       "\tsample14 & 12mo\\\\\n",
       "\tsample15 & 12mo\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 15 × 1\n",
       "\n",
       "| <!--/--> | condition &lt;fct&gt; |\n",
       "|---|---|\n",
       "| sample1 | 2mo  |\n",
       "| sample2 | 2mo  |\n",
       "| sample3 | 2mo  |\n",
       "| sample4 | 4mo  |\n",
       "| sample5 | 4mo  |\n",
       "| sample6 | 4mo  |\n",
       "| sample7 | 6mo  |\n",
       "| sample8 | 6mo  |\n",
       "| sample9 | 6mo  |\n",
       "| sample10 | 9mo  |\n",
       "| sample11 | 9mo  |\n",
       "| sample12 | 9mo  |\n",
       "| sample13 | 12mo |\n",
       "| sample14 | 12mo |\n",
       "| sample15 | 12mo |\n",
       "\n"
      ],
      "text/plain": [
       "         condition\n",
       "sample1  2mo      \n",
       "sample2  2mo      \n",
       "sample3  2mo      \n",
       "sample4  4mo      \n",
       "sample5  4mo      \n",
       "sample6  4mo      \n",
       "sample7  6mo      \n",
       "sample8  6mo      \n",
       "sample9  6mo      \n",
       "sample10 9mo      \n",
       "sample11 9mo      \n",
       "sample12 9mo      \n",
       "sample13 12mo     \n",
       "sample14 12mo     \n",
       "sample15 12mo     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 15 × 1</caption>\n",
       "<thead>\n",
       "\t<tr><th></th><th scope=col>condition</th></tr>\n",
       "\t<tr><th></th><th scope=col>&lt;fct&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><th scope=row>sample31</th><td>2mo </td></tr>\n",
       "\t<tr><th scope=row>sample32</th><td>2mo </td></tr>\n",
       "\t<tr><th scope=row>sample33</th><td>2mo </td></tr>\n",
       "\t<tr><th scope=row>sample34</th><td>4mo </td></tr>\n",
       "\t<tr><th scope=row>sample35</th><td>4mo </td></tr>\n",
       "\t<tr><th scope=row>sample36</th><td>4mo </td></tr>\n",
       "\t<tr><th scope=row>sample37</th><td>6mo </td></tr>\n",
       "\t<tr><th scope=row>sample38</th><td>6mo </td></tr>\n",
       "\t<tr><th scope=row>sample39</th><td>6mo </td></tr>\n",
       "\t<tr><th scope=row>sample40</th><td>9mo </td></tr>\n",
       "\t<tr><th scope=row>sample41</th><td>9mo </td></tr>\n",
       "\t<tr><th scope=row>sample42</th><td>9mo </td></tr>\n",
       "\t<tr><th scope=row>sample43</th><td>12mo</td></tr>\n",
       "\t<tr><th scope=row>sample44</th><td>12mo</td></tr>\n",
       "\t<tr><th scope=row>sample45</th><td>12mo</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 15 × 1\n",
       "\\begin{tabular}{r|l}\n",
       "  & condition\\\\\n",
       "  & <fct>\\\\\n",
       "\\hline\n",
       "\tsample31 & 2mo \\\\\n",
       "\tsample32 & 2mo \\\\\n",
       "\tsample33 & 2mo \\\\\n",
       "\tsample34 & 4mo \\\\\n",
       "\tsample35 & 4mo \\\\\n",
       "\tsample36 & 4mo \\\\\n",
       "\tsample37 & 6mo \\\\\n",
       "\tsample38 & 6mo \\\\\n",
       "\tsample39 & 6mo \\\\\n",
       "\tsample40 & 9mo \\\\\n",
       "\tsample41 & 9mo \\\\\n",
       "\tsample42 & 9mo \\\\\n",
       "\tsample43 & 12mo\\\\\n",
       "\tsample44 & 12mo\\\\\n",
       "\tsample45 & 12mo\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 15 × 1\n",
       "\n",
       "| <!--/--> | condition &lt;fct&gt; |\n",
       "|---|---|\n",
       "| sample31 | 2mo  |\n",
       "| sample32 | 2mo  |\n",
       "| sample33 | 2mo  |\n",
       "| sample34 | 4mo  |\n",
       "| sample35 | 4mo  |\n",
       "| sample36 | 4mo  |\n",
       "| sample37 | 6mo  |\n",
       "| sample38 | 6mo  |\n",
       "| sample39 | 6mo  |\n",
       "| sample40 | 9mo  |\n",
       "| sample41 | 9mo  |\n",
       "| sample42 | 9mo  |\n",
       "| sample43 | 12mo |\n",
       "| sample44 | 12mo |\n",
       "| sample45 | 12mo |\n",
       "\n"
      ],
      "text/plain": [
       "         condition\n",
       "sample31 2mo      \n",
       "sample32 2mo      \n",
       "sample33 2mo      \n",
       "sample34 4mo      \n",
       "sample35 4mo      \n",
       "sample36 4mo      \n",
       "sample37 6mo      \n",
       "sample38 6mo      \n",
       "sample39 6mo      \n",
       "sample40 9mo      \n",
       "sample41 9mo      \n",
       "sample42 9mo      \n",
       "sample43 12mo     \n",
       "sample44 12mo     \n",
       "sample45 12mo     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "month <- factor(rep(c(\"2mo\", \"4mo\", \"6mo\", \"9mo\", \"12mo\"), each = 3))\n",
    "# Change levels so that 2mo is the reference\n",
    "month <- factor(month, levels = c(\"2mo\", \"4mo\", \"6mo\", \"9mo\", \"12mo\" )) \n",
    "\n",
    "labelsM <- data.frame(condition = month, row.names = colnames(micro))\n",
    "labelsM\n",
    "\n",
    "labelsA <- data.frame(condition = month, row.names = colnames(astro))\n",
    "labelsA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "f1c05683",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "converting counts to integer mode\n",
      "\n",
      "estimating size factors\n",
      "\n",
      "estimating dispersions\n",
      "\n",
      "gene-wise dispersion estimates\n",
      "\n",
      "mean-dispersion relationship\n",
      "\n",
      "final dispersion estimates\n",
      "\n",
      "fitting model and testing\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'Intercept'</li><li>'condition_4mo_vs_2mo'</li><li>'condition_6mo_vs_2mo'</li><li>'condition_9mo_vs_2mo'</li><li>'condition_12mo_vs_2mo'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'Intercept'\n",
       "\\item 'condition\\_4mo\\_vs\\_2mo'\n",
       "\\item 'condition\\_6mo\\_vs\\_2mo'\n",
       "\\item 'condition\\_9mo\\_vs\\_2mo'\n",
       "\\item 'condition\\_12mo\\_vs\\_2mo'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'Intercept'\n",
       "2. 'condition_4mo_vs_2mo'\n",
       "3. 'condition_6mo_vs_2mo'\n",
       "4. 'condition_9mo_vs_2mo'\n",
       "5. 'condition_12mo_vs_2mo'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"Intercept\"             \"condition_4mo_vs_2mo\"  \"condition_6mo_vs_2mo\" \n",
       "[4] \"condition_9mo_vs_2mo\"  \"condition_12mo_vs_2mo\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Run DESeq() on the microglia data\n",
    "ddsM <- DESeqDataSetFromMatrix(countData = round(micro), colData = labelsM, design = ~ condition)\n",
    "ddsM <- DESeq(ddsM)\n",
    "resultsNames(ddsM) # gives coefficients"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "879fa063",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "\n",
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "\n",
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "\n",
      "using 'apeglm' for LFC shrinkage. If used in published research, please cite:\n",
      "    Zhu, A., Ibrahim, J.G., Love, M.I. (2018) Heavy-tailed prior distributions for\n",
      "    sequence count data: removing the noise and preserving large differences.\n",
      "    Bioinformatics. https://doi.org/10.1093/bioinformatics/bty895\n",
      "\n"
     ]
    }
   ],
   "source": [
    "# Get the results for each of the 4 microglia comparisons (with log fold change shrunken) \n",
    "res4M <- lfcShrink(ddsM, coef=\"condition_4mo_vs_2mo\", type=\"apeglm\")\n",
    "res6M <- lfcShrink(ddsM, coef=\"condition_6mo_vs_2mo\", type=\"apeglm\")\n",
    "res9M <- lfcShrink(ddsM, coef=\"condition_9mo_vs_2mo\", type=\"apeglm\")\n",
    "res12M <- lfcShrink(ddsM, coef=\"condition_12mo_vs_2mo\", type=\"apeglm\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "d04f6073",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "log2 fold change (MAP): condition 4mo vs 2mo \n",
       "Wald test p-value: condition 4mo vs 2mo \n",
       "DataFrame with 61 rows and 5 columns\n",
       "               baseMean log2FoldChange     lfcSE      pvalue        padj\n",
       "              <numeric>      <numeric> <numeric>   <numeric>   <numeric>\n",
       "2010107G23Rik   70.4262       5.592144  0.818035 1.10732e-12 8.33587e-09\n",
       "4930522L14Rik  229.2585       0.788272  0.202055 4.17571e-06 2.41806e-03\n",
       "ATP8           178.2856       0.855979  0.272844 6.48576e-05 1.90161e-02\n",
       "Adgrl4          28.5871       2.290025  0.776773 1.00593e-04 2.40402e-02\n",
       "Anp32a         792.9677       1.101585  0.252435 5.31547e-07 6.15613e-04\n",
       "...                 ...            ...       ...         ...         ...\n",
       "Wfdc1           14.9516       2.972623  1.125206 2.34198e-04 4.22278e-02\n",
       "Zcchc9         242.9688       0.547850  0.197283 2.30004e-04 4.22278e-02\n",
       "Zfp69          964.6476       0.587253  0.113520 1.24530e-08 2.67846e-05\n",
       "Zfp738         224.5707       0.817991  0.266217 8.28068e-05 2.14955e-02\n",
       "Zfyve16        255.6202       0.539743  0.186831 1.67570e-04 3.60420e-02"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "log2 fold change (MAP): condition 6mo vs 2mo \n",
       "Wald test p-value: condition 6mo vs 2mo \n",
       "DataFrame with 1738 rows and 5 columns\n",
       "               baseMean log2FoldChange     lfcSE      pvalue        padj\n",
       "              <numeric>      <numeric> <numeric>   <numeric>   <numeric>\n",
       "0610007P14Rik   407.622       0.515548  0.138260 7.99402e-05 1.60409e-03\n",
       "0610010B08Rik   104.861      -2.352085  1.384025 2.15151e-03 2.20291e-02\n",
       "0610012G03Rik   669.195       0.535978  0.128827 1.23812e-05 3.52113e-04\n",
       "1110004F10Rik  1223.964       0.553751  0.114589 5.05115e-07 2.31202e-05\n",
       "1110037F02Rik   403.185      -1.277463  0.304066 2.37521e-06 8.83585e-05\n",
       "...                 ...            ...       ...         ...         ...\n",
       "Znrf3           67.7973      -1.372337  0.302261 4.96658e-07 2.28006e-05\n",
       "Zrsr1          164.5006       1.499184  0.611024 7.74263e-04 9.85628e-03\n",
       "Zswim8         265.9064      -1.179536  0.359021 9.30949e-05 1.82544e-03\n",
       "Zyg11b         352.1839      -0.680536  0.209245 2.71596e-04 4.27453e-03\n",
       "Zzef1          336.1697      -1.429111  0.318899 5.89133e-07 2.64188e-05"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "log2 fold change (MAP): condition 9mo vs 2mo \n",
       "Wald test p-value: condition 9mo vs 2mo \n",
       "DataFrame with 121 rows and 5 columns\n",
       "                baseMean log2FoldChange     lfcSE      pvalue        padj\n",
       "               <numeric>      <numeric> <numeric>   <numeric>   <numeric>\n",
       "1810011O10Rik   253.3230       1.740802  0.344540 1.82203e-08 2.28780e-05\n",
       "2010107G23Rik    70.4262       4.807224  0.823745 5.85584e-10 1.01101e-06\n",
       "2410006H16Rik   220.2610       0.729186  0.270570 2.67978e-04 2.99933e-02\n",
       "6330407A03Rik   302.6811       0.723659  0.246269 1.33479e-04 1.72301e-02\n",
       "Arhgap15         71.5772       2.290797  0.625287 8.92423e-06 2.20110e-03\n",
       "...                  ...            ...       ...         ...         ...\n",
       "Tnfaip3       2284.82143       0.873028  0.257270 2.64882e-05 0.005302249\n",
       "Tslp             7.54977       3.031726  1.100491 2.39804e-04 0.027833348\n",
       "Umodl1          28.81206       4.805304  1.350474 2.11206e-05 0.004289962\n",
       "Wscd1          147.97040       1.304028  0.318930 1.80502e-06 0.000644311\n",
       "Zfp69          964.64765       0.502257  0.114039 6.44127e-07 0.000326660"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "log2 fold change (MAP): condition 12mo vs 2mo \n",
       "Wald test p-value: condition 12mo vs 2mo \n",
       "DataFrame with 1259 rows and 5 columns\n",
       "               baseMean log2FoldChange     lfcSE      pvalue        padj\n",
       "              <numeric>      <numeric> <numeric>   <numeric>   <numeric>\n",
       "0610010B08Rik  104.8609        2.70886  1.179185 6.19781e-04 1.47857e-02\n",
       "0610030E20Rik  905.8935        1.07929  0.583245 4.66946e-03 4.78092e-02\n",
       "0610039K10Rik   19.8482        2.33408  0.979386 5.63779e-04 1.39995e-02\n",
       "1110038F14Rik  415.2761        1.09878  0.382391 3.85664e-04 1.10266e-02\n",
       "1600020E01Rik  137.2636        1.51349  0.313356 9.65792e-08 2.38206e-05\n",
       "...                 ...            ...       ...         ...         ...\n",
       "Zmym6           483.432       0.752596  0.322186 0.003373650  0.04002468\n",
       "Znrf1           847.085       0.974706  0.449110 0.002983384  0.03715645\n",
       "Zrsr2           302.681       1.180264  0.389205 0.000210772  0.00757234\n",
       "Zscan26         645.429       1.259390  0.533494 0.001255256  0.02290024\n",
       "Zzz3            425.973       0.803754  0.331035 0.002338829  0.03272939"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# DEGs defined as genes with padj < 0.05 and |log2FoldChange| > 0.5\n",
    "# Filter for both (and get rid of any NA values) for each microglia comparison\n",
    "res4M_filt <- subset(res4M, padj != \"NA\" & padj < 0.05)\n",
    "res4M_filt <- subset(res4M_filt, log2FoldChange != \"NA\" & (log2FoldChange > 0.5 | log2FoldChange < -0.5))\n",
    "res4M_filt\n",
    "\n",
    "res6M_filt <- subset(res6M, padj != \"NA\" & padj < 0.05)\n",
    "res6M_filt <- subset(res6M_filt, log2FoldChange != \"NA\" & (log2FoldChange > 0.5 | log2FoldChange < -0.5))\n",
    "res6M_filt\n",
    "\n",
    "res9M_filt <- subset(res9M, padj != \"NA\" & padj < 0.05)\n",
    "res9M_filt <- subset(res9M_filt, log2FoldChange != \"NA\" & (log2FoldChange > 0.5 | log2FoldChange < -0.5))\n",
    "res9M_filt\n",
    "\n",
    "res12M_filt <- subset(res12M, padj != \"NA\" & padj < 0.05)\n",
    "res12M_filt <- subset(res12M_filt, log2FoldChange != \"NA\" & (log2FoldChange > 0.5 | log2FoldChange < -0.5))\n",
    "res12M_filt"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "2e7d1339",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "61"
      ],
      "text/latex": [
       "61"
      ],
      "text/markdown": [
       "61"
      ],
      "text/plain": [
       "[1] 61"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'2010107G23Rik'</li><li>'4930522L14Rik'</li><li>'ATP8'</li><li>'Adgrl4'</li><li>'Anp32a'</li><li>'Arl6ip5'</li><li>'Atp6v0a2'</li><li>'COX2'</li><li>'Ccdc82'</li><li>'Chst15'</li><li>'Csmd3'</li><li>'Cxcl10'</li><li>'Cxcl14'</li><li>'Cyp2j9'</li><li>'Dact3'</li><li>'Dhx9'</li><li>'Edn1'</li><li>'Egfl8'</li><li>'Eps8l1'</li><li>'Fam178a'</li><li>'Fmr1'</li><li>'Gm14023'</li><li>'Gm26520'</li><li>'Gm31546'</li><li>'Gm36365'</li><li>'Gm40346'</li><li>'Gm41964'</li><li>'Gpr19'</li><li>'H2-Q1'</li><li>'Ifi44'</li><li>'Igfbp7'</li><li>'Inppl1'</li><li>'Kif11'</li><li>'Ly6c1'</li><li>'Map2k4'</li><li>'Mcm3'</li><li>'Med12l'</li><li>'Midn'</li><li>'Mki67'</li><li>'Mlxipl'</li><li>'Msh5'</li><li>'Mt3'</li><li>'ND1'</li><li>'ND4L'</li><li>'ND5'</li><li>'Nectin2'</li><li>'Pa2g4'</li><li>'Pbk'</li><li>'Pde2a'</li><li>'Ptpn23'</li><li>'Rps27'</li><li>'Sapcd1'</li><li>'Scoc'</li><li>'Slfn8'</li><li>'Tmco1'</li><li>'Upk1b'</li><li>'Wfdc1'</li><li>'Zcchc9'</li><li>'Zfp69'</li><li>'Zfp738'</li><li>'Zfyve16'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '2010107G23Rik'\n",
       "\\item '4930522L14Rik'\n",
       "\\item 'ATP8'\n",
       "\\item 'Adgrl4'\n",
       "\\item 'Anp32a'\n",
       "\\item 'Arl6ip5'\n",
       "\\item 'Atp6v0a2'\n",
       "\\item 'COX2'\n",
       "\\item 'Ccdc82'\n",
       "\\item 'Chst15'\n",
       "\\item 'Csmd3'\n",
       "\\item 'Cxcl10'\n",
       "\\item 'Cxcl14'\n",
       "\\item 'Cyp2j9'\n",
       "\\item 'Dact3'\n",
       "\\item 'Dhx9'\n",
       "\\item 'Edn1'\n",
       "\\item 'Egfl8'\n",
       "\\item 'Eps8l1'\n",
       "\\item 'Fam178a'\n",
       "\\item 'Fmr1'\n",
       "\\item 'Gm14023'\n",
       "\\item 'Gm26520'\n",
       "\\item 'Gm31546'\n",
       "\\item 'Gm36365'\n",
       "\\item 'Gm40346'\n",
       "\\item 'Gm41964'\n",
       "\\item 'Gpr19'\n",
       "\\item 'H2-Q1'\n",
       "\\item 'Ifi44'\n",
       "\\item 'Igfbp7'\n",
       "\\item 'Inppl1'\n",
       "\\item 'Kif11'\n",
       "\\item 'Ly6c1'\n",
       "\\item 'Map2k4'\n",
       "\\item 'Mcm3'\n",
       "\\item 'Med12l'\n",
       "\\item 'Midn'\n",
       "\\item 'Mki67'\n",
       "\\item 'Mlxipl'\n",
       "\\item 'Msh5'\n",
       "\\item 'Mt3'\n",
       "\\item 'ND1'\n",
       "\\item 'ND4L'\n",
       "\\item 'ND5'\n",
       "\\item 'Nectin2'\n",
       "\\item 'Pa2g4'\n",
       "\\item 'Pbk'\n",
       "\\item 'Pde2a'\n",
       "\\item 'Ptpn23'\n",
       "\\item 'Rps27'\n",
       "\\item 'Sapcd1'\n",
       "\\item 'Scoc'\n",
       "\\item 'Slfn8'\n",
       "\\item 'Tmco1'\n",
       "\\item 'Upk1b'\n",
       "\\item 'Wfdc1'\n",
       "\\item 'Zcchc9'\n",
       "\\item 'Zfp69'\n",
       "\\item 'Zfp738'\n",
       "\\item 'Zfyve16'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '2010107G23Rik'\n",
       "2. '4930522L14Rik'\n",
       "3. 'ATP8'\n",
       "4. 'Adgrl4'\n",
       "5. 'Anp32a'\n",
       "6. 'Arl6ip5'\n",
       "7. 'Atp6v0a2'\n",
       "8. 'COX2'\n",
       "9. 'Ccdc82'\n",
       "10. 'Chst15'\n",
       "11. 'Csmd3'\n",
       "12. 'Cxcl10'\n",
       "13. 'Cxcl14'\n",
       "14. 'Cyp2j9'\n",
       "15. 'Dact3'\n",
       "16. 'Dhx9'\n",
       "17. 'Edn1'\n",
       "18. 'Egfl8'\n",
       "19. 'Eps8l1'\n",
       "20. 'Fam178a'\n",
       "21. 'Fmr1'\n",
       "22. 'Gm14023'\n",
       "23. 'Gm26520'\n",
       "24. 'Gm31546'\n",
       "25. 'Gm36365'\n",
       "26. 'Gm40346'\n",
       "27. 'Gm41964'\n",
       "28. 'Gpr19'\n",
       "29. 'H2-Q1'\n",
       "30. 'Ifi44'\n",
       "31. 'Igfbp7'\n",
       "32. 'Inppl1'\n",
       "33. 'Kif11'\n",
       "34. 'Ly6c1'\n",
       "35. 'Map2k4'\n",
       "36. 'Mcm3'\n",
       "37. 'Med12l'\n",
       "38. 'Midn'\n",
       "39. 'Mki67'\n",
       "40. 'Mlxipl'\n",
       "41. 'Msh5'\n",
       "42. 'Mt3'\n",
       "43. 'ND1'\n",
       "44. 'ND4L'\n",
       "45. 'ND5'\n",
       "46. 'Nectin2'\n",
       "47. 'Pa2g4'\n",
       "48. 'Pbk'\n",
       "49. 'Pde2a'\n",
       "50. 'Ptpn23'\n",
       "51. 'Rps27'\n",
       "52. 'Sapcd1'\n",
       "53. 'Scoc'\n",
       "54. 'Slfn8'\n",
       "55. 'Tmco1'\n",
       "56. 'Upk1b'\n",
       "57. 'Wfdc1'\n",
       "58. 'Zcchc9'\n",
       "59. 'Zfp69'\n",
       "60. 'Zfp738'\n",
       "61. 'Zfyve16'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"2010107G23Rik\" \"4930522L14Rik\" \"ATP8\"          \"Adgrl4\"       \n",
       " [5] \"Anp32a\"        \"Arl6ip5\"       \"Atp6v0a2\"      \"COX2\"         \n",
       " [9] \"Ccdc82\"        \"Chst15\"        \"Csmd3\"         \"Cxcl10\"       \n",
       "[13] \"Cxcl14\"        \"Cyp2j9\"        \"Dact3\"         \"Dhx9\"         \n",
       "[17] \"Edn1\"          \"Egfl8\"         \"Eps8l1\"        \"Fam178a\"      \n",
       "[21] \"Fmr1\"          \"Gm14023\"       \"Gm26520\"       \"Gm31546\"      \n",
       "[25] \"Gm36365\"       \"Gm40346\"       \"Gm41964\"       \"Gpr19\"        \n",
       "[29] \"H2-Q1\"         \"Ifi44\"         \"Igfbp7\"        \"Inppl1\"       \n",
       "[33] \"Kif11\"         \"Ly6c1\"         \"Map2k4\"        \"Mcm3\"         \n",
       "[37] \"Med12l\"        \"Midn\"          \"Mki67\"         \"Mlxipl\"       \n",
       "[41] \"Msh5\"          \"Mt3\"           \"ND1\"           \"ND4L\"         \n",
       "[45] \"ND5\"           \"Nectin2\"       \"Pa2g4\"         \"Pbk\"          \n",
       "[49] \"Pde2a\"         \"Ptpn23\"        \"Rps27\"         \"Sapcd1\"       \n",
       "[53] \"Scoc\"          \"Slfn8\"         \"Tmco1\"         \"Upk1b\"        \n",
       "[57] \"Wfdc1\"         \"Zcchc9\"        \"Zfp69\"         \"Zfp738\"       \n",
       "[61] \"Zfyve16\"      "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "DEG_4M <- row.names(res4M_filt)\n",
    "DEG_6M <- row.names(res4M_filt)\n",
    "DEG_9M <- row.names(res4M_filt)\n",
    "DEG_12M <- row.names(res4M_filt)\n",
    "\n",
    "# Find microglia DEGs that occur at all time points\n",
    "DEG_allM <- Reduce(intersect, list(DEG_4M, DEG_6M, DEG_9M, DEG_12M))\n",
    "length(DEG_allM)\n",
    "DEG_allM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3ac920cd",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
